src/figuresClasses.ts

```diff
@@ -1,19 +1,86 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape
+  color: Color
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (
+      a <= 0
+      || b <= 0
+      || c <= 0
+      || a + b <= c
+      || b + c <= a
+      || a + c <= b
+    ) {
+      throw new Error('invalid numbers');
```

Comment:
Checklist p3
How should I know after this error what are valid?

Reply:
Was fixed